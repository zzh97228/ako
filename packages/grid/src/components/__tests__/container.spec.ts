import { mount, VueWrapper } from '@vue/test-utils';
import Container from '../container';

describe('container.ts', () => {
  let mountFunc: (opts?: object) => VueWrapper<any>;
  beforeEach(() => {
    mountFunc = (options = {}) => mount(Container, options);
  });
  it('should render container with customized tag and size', () => {
    const wrapper = mountFunc({
      props: {
        tag: 'main',
        height: '200px',
        width: '100px',
      },
      slots: {
        default: () => 'container',
      },
    });

    expect(wrapper.classes()).toContain('container');
    expect(wrapper.attributes('style')).toContain('width: 100px');
    expect(wrapper.attributes('style')).toContain('height: 200px');
    expect(wrapper.element.tagName).toEqual('MAIN');
    expect(wrapper.text()).toBe('container');
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should have fluid classes when set fluid props', () => {
    const wrapper = mountFunc({
      props: {
        fluid: true,
      },
    });

    expect(wrapper.classes()).toContain('container--fluid');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
