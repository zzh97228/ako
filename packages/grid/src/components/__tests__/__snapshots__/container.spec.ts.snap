// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container.ts should render container with customized tag and size 1`] = `
<main class="container"
      style="width: 100px; height: 200px;"
>
</main>
`;
